src/components/PhotoList.jsx

```diff
@@ -1,30 +1,62 @@
 import styles from '../styles/photoList'
 
-import React from 'react'
+import React, { Component } from 'react'
+import Dimensions from 'react-dimensions'
+
+import justifiedLayout from 'justified-layout'
 
 import Empty from './Empty'
 import Photo from './Photo'
 
 import classNames from 'classnames'
 
-export const PhotoList = ({ key, title, photos, selected, onPhotoToggle }) => {
-  return (
-    <div
-      className={classNames(styles['pho-section'], selected.length && styles['pho-section--has-selection'])}
-      key={key}
-      >
-      {!!title && <h3>{title}</h3>}
-      {photos.map(photo =>
-        <Photo
-          photo={photo}
-          key={photo._id}
-          selected={selected.indexOf(photo._id) !== -1}
-          onToggle={onPhotoToggle}
-        />
-      )}
-      {photos.length === 0 && <Empty emptyType='photos' />}
-    </div>
-  )
+const photoDimensionsFallback = {width: 1, height: 1}
+
+// Returns pseudo responsive row height based on container width. Trying to get
+// something between 190 and 240.
+const adaptRowHeight = containerWidth => 180 + ((containerWidth || 1800) / 30)
+
+export class PhotoList extends Component {
+  render () {
+    const { key, title, photos, selected, onPhotoToggle, containerWidth } = this.props
+    // @see https://flickr.github.io/justified-layout/
+    const layout = justifiedLayout(
+      photos.map(photo => photo.metadata || photoDimensionsFallback),
+      {
+        containerWidth: containerWidth,
+        targetRowHeight: adaptRowHeight(containerWidth),
+        // Must be relevant with styles
+        boxSpacing: {
+          horizontal: 16,
```

Comment:
Yep, it made me sick too, but I did not figure out a quick way to have something clean. I am totally open to suggestions 😭